fission-web-server/library/Fission/Web/Server/Types.hs

```diff
@@ -373,25 +372,44 @@ instance (Eq a, Display a) => MonadIPFSCluster Server a where
 
       asyncRef <- liftIO $ async do
         Stream.withClientM streamQuery (mkClientEnv ipfsHttpManager url) \event ->
-          runServer cfg $
```

Comment:
For reference: `runServer :: Config -> Fission.Server a -> IO a`. So it turns "stuff that knows rich info about our server setup" into "hey it's running some generic IO for reals" (which is _always_ the pattern).

This got pushed down into each branch, mostly because `foreach` needs to return `IO ()` for the success and failure conditions (we could have left it here and done a bunch more `liftIO`, but it was ugly). Remember that this is happening in a stream off of a real world HLS connection, which means that `IO` is happening in here in general (they _could_ have made it a `MonadIO m` but didn't).

We totally _could_ make this work directly in `Fission.Server`, but we're already in `IO` because of that `async` on line 373 forking a process, so this is easier.